### PR TITLE
Idea for movement restrictions

### PR DESCRIPTION
**Pull request by dagal10**
I think a lot more rules could be included with this, but as an example this can prevent miss clicking. Making it a little easier to stay within the tiles.

### CODE REVIEW

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 1]
```diff
@@ -616,6 +624,19 @@ int getXpUntilNextTile() {
         return xpUntilNextTile;
     }
 
+    private boolean modeTilescontainSelectedSceneTile(){
```
Comment (ConorLeckey) [inline src/main/java/com/tileman/TilemanModePlugin.java:627]: A few things, if you could rename to 'isTileMarked' and add a parameter 'tile'. Just means we can use this elsewhere if we want to know if any tile is marked.

File: src/main/java/com/tileman/TilemanModePlugin.java [hunk 2]
```diff
@@ -131,6 +131,14 @@ TilemanModeConfig provideConfig(ConfigManager configManager) {
     public void onMenuOptionClicked(MenuOptionClicked event) {
         if (event.getMenuAction().getId() != MenuAction.RUNELITE.getId() ||
                 !(event.getMenuOption().equals(MARK) || event.getMenuOption().equals(UNMARK))) {
+            if(event.getMenuAction() == MenuAction.WALK && !modeTilescontainSelectedSceneTile() ){
+                //We are attempting to walking to a location and it is not in our list
+                if(!config.automarkTiles() || (config.automarkTiles() && remainingTiles == 0)) {
```
Comment (ConorLeckey) [inline src/main/java/com/tileman/TilemanModePlugin.java:136]: I'm thinking this functionality should be optional, could you add a config option?

**Review by ConorLeckey — COMMENTED** [review]
Looks great so far! I've left some comments :)

**Comment by dagal10** [comment]
Sorry, I just realized that this should have been a draft.
I wanted to make sure that this was something you would be interested in before any more time was spent on it.

**Comment by ConorLeckey** [comment]
> Sorry, I just realized that this should have been a draft.
> I wanted to make sure that this was something you would be interested in before any more time was spent on it.

I love the idea personally :) @sram1337 what do you think?

**Comment by dagal10** [comment]
I made the changes you wanted, added more in for reusability, and updated existing code to use those methods.
I don't normally write in Java so if I am not doing things the Java way let me know and I can fix it.

**Comment by dagal10** [comment]
Well.... don't do anything with this yet.
My account has been banned for macroing.
No botting has been done on this account (It is my Chunkman mode account that has already gotten 60 wood cutting, so this is disappointing.)

**Comment by ConorLeckey** [comment]
Damn that sucks man, I was worried about something like this happening but wasnt sure. Going to close this MR

**Pull request closed without merging**